### PR TITLE
feat: add `--only` flag in `cosign copy` to copy sign, att & sbom

### PR DESCRIPTION
**Pull request by vishal-chdhry**
#### Summary

Closes: #2002

This PR will add a new `--only` flag that accept custom string array to download sboms, signatures and attestations
```bash
$ cosign copy --only=sbom,sign,att <SRC> <DST>
```
#### Release Note

Added `--only` flag in `cosign copy` to copy sign, att & sbom
#### Documentation

Ran `make docgen` to generate required docs

### CODE REVIEW

File: cmd/cosign/cli/options/copy.go [hunk 1]
```diff
@@ -33,8 +34,11 @@ var _ Interface = (*CopyOptions)(nil)
 func (o *CopyOptions) AddFlags(cmd *cobra.Command) {
 	o.Registry.AddFlags(cmd)
 
+	cmd.Flags().StringVar(&o.CopyOnly, "only", "",
+		"custom string array to only copy specific items. ex: --only=sbom,sign,att")
```
Comment (haydentherapper) [inline cmd/cosign/cli/options/copy.go:38]: can you note "comma delimited" in the description?

File: cmd/cosign/cli/copy/copy.go [hunk 2]
```diff
@@ -130,8 +126,8 @@ func CopyCmd(ctx context.Context, regOpts options.RegistryOptions, srcImg, dstIm
 		return err
 	}
 
-	// If we're only copying sigs, we have nothing left to do.
-	if sigOnly {
+	// If we're only copying sig/att/sbom, we have nothing left to do.
+	if len(parseOnlyFlag(copyOnly, sigOnly)) > 0 {
```
Comment (haydentherapper) [inline cmd/cosign/cli/copy/copy.go:130]: Can we store the results of the previous call rather than call this again?

**Review by haydentherapper — CHANGES_REQUESTED** [review]
Looks good, just a few nits!

Sorry for the delay, working through the backlog of PRs.

**Comment by vishal-chdhry** [comment]
@haydentherapper Fixed, can you please re-review

File: cmd/cosign/cli/copy/copy.go [hunk 3]
```diff
@@ -175,12 +174,13 @@ func remoteCopy(ctx context.Context, pusher *remote.Pusher, src, dest name.Refer
 	return pusher.Push(ctx, dest, got)
 }
 
-func parseOnlyFlag(str string, sigOnly bool) []tagMap {
+func parseeOnlyOpt(str string, sigOnly bool) []tagMap {
```
Comment (haydentherapper) [inline cmd/cosign/cli/copy/copy.go:177]: spelling "parseOnlyOpt"

**Review by haydentherapper — COMMENTED** [review]
Can you also run `make docgen`?

**Review by haydentherapper — APPROVED** [review]

**Comment by vishal-chdhry** [comment]
@haydentherapper fixed

**Pull request merged**